otherlibs/unix/read_unix.c

```diff
@@ -35,3 +36,21 @@ CAMLprim value caml_unix_read(value fd, value buf, value ofs, value len)
   memmove (&Byte(buf, Long_val(ofs)), iobuf, ret);
   CAMLreturn(Val_int(ret));
 }
+
+CAMLprim value caml_unix_read_bigarray(value fd, value vbuf,
+                                       value vofs, value vlen)
+{
+  CAMLparam4(fd, vbuf, vofs, vlen);
```

Comment:
Up to you, but there's no need to register things that are required to be integers with the GC:
```suggestion
  CAMLparam2(fd, vbuf);
```
(equally, that thinking is how the `fd` wasn't registered as a root...)

Reply:
See https://github.com/ocaml/ocaml/pull/12365#discussion_r1263491872